### ISSUE TITLE
Bug Report: unable to connect remote Replica Set connect unless proxy it on 127.0.0.1

### ISSUE THREAD

**Issue by Fallsleep** [mongodb-js/vscode#564]
#### Current Behavior

When I try to connect to the replica set mongodb in `VirtulaBox`，it come into `Unable to connect: connect ECONNREFUSED 127.0.0.1:27017`. 

connect string: `mongodb://homestead:secret@192.168.10.10:27017,192.168.10.10:27018,192.168.10.10:27019/?authSource=admin&replicaSet=rs0&readPreference=primary&ssl=false`

after setting `Port Forwarding` rule in `VirtulaBox`, it connect well.

I guess there're some bugs with replica set connect code.

#### Environment

- node.js / npm versions:
- OS:

**Comment by harshita-2003** [comment]
![Screenshot 2023-10-18 172138](https://github.com/mongodb-js/vscode/assets/119532943/65188355-55fb-4679-b152-f90d8df6a816)

I'm getting in mongodb compass ,how to resolve this though this is not a compass issue

**Comment by addaleax** [comment]
@Fallsleep Sorry, this went under our radar. You probably configured your replica sets to refer to each other using the `127.0.0.1` address, the client (regardless of which MongoDB client you are using) detects that and then tries to connect to that endpoint as well (because it doesn’t know that `127.0.0.1:27017` and `192.168.10.10:27017` refer to the same address).

You can either use `directConnection=true` to connect to a single host to work around this, or you could set up a `mongos` in front of the replica set and then connect to that.

@harshita-2003 Yeah, that’s probably a separate issue. Most likely, your MongoDB server is not actually running on port 27017; that’s what you’d want to check first.

**Comment by harshita-2003** [comment]
It worked after start the mongodb from services , thankyou!

**Comment by Fallsleep** [comment]
Yes, I changed the replica set config to the virtual box host name instead of 127.0.0.1, and add the host name in hosts file, it can work well.
